src/app/InteractionModelEngine.cpp

```diff
@@ -314,6 +313,15 @@ CHIP_ERROR InteractionModelEngine::OnReadInitialRequest(Messaging::ExchangeConte
         }
     }
 
+    // Reserve the last ReadHandler for ReadInteraction
+    if (aInteractionType == ReadHandler::InteractionType::Subscribe &&
+        ((CHIP_IM_MAX_NUM_READ_HANDLER - GetNumActiveReadHandlers()) == 1))
```

Comment:
This doesn't account for the possibility that the other handlers in use might be for reads and not subscribes right? In which case, we do not want to reject this subscription?

Reply:
agree, just updated with new check for "bool InteractionModelEngine::HasActiveRead() const"

Reply:
Done，thanks